### PR TITLE
Add Getting Started info to README.md

### PR DESCRIPTION
**Pull request by louwers**
This PR adds some information to get started with MapLibre Native in the main `README.md`.

I have gotten some feedback that users expect this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,42 +14,196 @@ This project originated as a fork of Mapbox GL Native, before their switch to a
...
 ## Contributing
 
-To contribute to MapLibre Native, see [`CONTRIBUTING.md`](CONTRIBUTING.md) and (if applicable) the specific instructions for the platform you want to contribute to.
+> [!NOTE]  
+> This section is only relevant for people who want to contribute to MapLibre Native.
+
+To get started with the code base, you need to clone the the repository including all its submodules.
+
+```bash
+git clone --recurse-submodules https://github.com/maplibre/maplibre-native.git
```
Comment (cgalvan) [inline README.md:182]: Should we recommend they create a fork first if they are going to contribute, so they can push their branch to it and ultimately create a PR?

File: README.md [hunk 2]
```diff
@@ -14,42 +14,196 @@ This project originated as a fork of Mapbox GL Native, before their switch to a
...
+
+## Android
+
+Open `platform/android` with Android Studio.
+
+More information: [`platform/android/DEVELOPING.md`](platform/android/DEVELOPING.md).
+
+## iOS
+
+You need to have
```
Comment (cgalvan) [inline README.md:198]: Can this line be removed or was there a specific pre-req you wanted to mention in this sentence before the "More information" link below it?

File: README.md (hunk 1 above)
Comment (louwers) [reply README.md:182]: Good idea!

File: README.md (hunk 2 above)
Comment (louwers) [reply README.md:198]: Oops thanks.

**Pull request merged**